### PR TITLE
propagate keyword argument:  normalize_to_only_use_kwargs of node.normalized_arguments function to normalize_function and normalize_module

### PR DESCRIPTION
**Pull request by asalioufb**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62894

### CODE REVIEW

File: test/test_fx.py [hunk 1]
```diff
@@ -2812,7 +2812,45 @@ def f(b, a):
...
         self.assertEqual(nf(**val), f(**val))
 
+    def test_node_normalize_arguments_use_kwargs(self):
+        class Tracer(torch.fx.Tracer):
+            def is_leaf_module(self, m, module_qualified_name):
+                return False
+
+        foo = torch.nn.Linear(4, 4)
+        trace = Tracer().trace(foo)
+        node = list(trace.nodes)[3]
```
Comment (jamesr66a) [inline test/test_fx.py:2822]: Here and below, we shouldn't make an assumption about the absolute position of the node of interest in the node list, as this may change due to unrelated changes in IR.

Instead, let's do something like this:
```
found_node = False
for node in trace.nodes:
  if node.target == torch.nn.functional.linear:
    found_node = True
    sig = inspect.signature(node.target)
    expected_result = {k: v for (k, v) in zip(sig.parameters, node.args)}
    expected_result.update(node.kwargs)
    self.assertEqual(output.kwargs, expected_result)
    self.assertEqual(len(output.args), 0)

self.assertTrue(found_node)
```

**Review by jamesr66a — APPROVED** [review]
Looks great! Small comment about testing discipline but this looks good to me

**Comment by asalioufb** [comment]
New PR was created even if last commit was amended
